presto-hive/src/test/java/com/facebook/presto/hive/TestHiveManifestUtils.java

```diff
@@ -76,6 +82,45 @@ public void testCreatePartitionManifest()
         assertEquals(manifestPage.get().getPositionCount(), 1);
     }
 
+    @Test
+    public void testFileNamesCompression()
+    {
+        List<String> fileNames = new ArrayList<>();
+
+        // File names are continuous and sequential
+        for (int i = 0; i <= 100; i++) {
+            fileNames.add(String.valueOf(i));
+        }
+
+        String compressedSequentialFileNames = compressFileNames(fileNames);
+        assertEquals(fileNames, decompressFileNames(compressedSequentialFileNames));
+
+        // File names NOT sequential
+        fileNames.clear();
+        for (int i = 0; i <= 100; i++) {
+            if (i % 10 == 0) {
+                continue;
+            }
+            fileNames.add(String.valueOf(i));
+        }
+
+        String compressedNonSequentialFileNames = compressFileNames(fileNames);
+        assertEquals(fileNames, decompressFileNames(compressedNonSequentialFileNames));
+    }
```

Comment:
Maybe also add one after this by randomly generating a sequence of numbers. Same like the one you did in `testFileSizesCompression`